### PR TITLE
DWW Document new Content-Type parameter for get/list user APIs

### PR DESCRIPTION
**Pull request by daytonwilliams-okta**
Resolves: OKTA-388384

## Description:
New Content-Type parameter added to allos omission of parts of certain User API responses that are expensive to compute for some configurations.

- **Is this PR related to a Monolith release?** 2021.04.2

### Resolves:

* [OKTA-388384](https://oktainc.atlassian.net/browse/OKTA-388384)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 1]
```diff
@@ -730,6 +730,29 @@ Fetches a user from your Okta organization
 - [Get User with Login](#get-user-with-login)
 - [Get User with Login Shortname](#get-user-with-login-shortname)
 
+##### Content-Type Parameters
+
+All variants of this API support a performance optimization.  Complex DelAuth configurations may degrade performance when fetching specific parts of the response.  Passing an optional Content-Type parameter can omit these parts, bypassing the bottleneck.
+
+| Parameter     | Description                                                                                                                           |
+| ------------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| okta-response | A comma-separated list of omit options (optionally surrounded in quotes) each specifying a specific part of the resonse to be omitted |
```
Comment (donwoods-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:739]: `resonse` --> `response`
also, can we come up with a less redundant phrasing than "specifying a specific part"?

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 2]
```diff
@@ -730,6 +730,29 @@ Fetches a user from your Okta organization
...
+
+| Omit Option | Description                                                                                                                     |
+| ----------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| omitCredentials | Omits the credentials subobject from the response.  |
+| omitCredentialsLinks | Response omits the following HAL links:  Change Password, Change Recovery Question, Forgot Password, Reset Password, Reset Factors, Unlock |
+| omitTransitioningToStatus | Omits the transitioningToStatus field from the response. |
```
Comment (donwoods-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:745]: Why not use the same format for each Description?  E.g. middle could be "Omits the following HAL links from the response: ...", or first could be "Response omits the credentials subobject."
nit: Middle item should end with period after list of links.

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 3]
```diff
@@ -1070,6 +1093,31 @@ Lists users in your organization with pagination in most cases
...
 A subset of users can be returned that match a supported filter expression or search criteria.
 
+
+##### Content-Type Parameters
+
+This API supports a performance optimization.  Complex DelAuth configurations may degrade performance when fetching specific parts of the response.  Passing an optional Content-Type parameter can omit these parts, bypassing the bottleneck.
+
+| Parameter     | Description                                                                                                                           |
+| ------------- | ------------------------------------------------------------------------------------------------------------------------------------- |
+| okta-response | A comma-separated list of omit options (optionally surrounded in quotes) each specifying a specific part of the resonse to be omitted |
```
Comment (donwoods-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:1103]: Same comments as above: `resonse`, "specify specific", inconsistent Description phrasing

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md (hunk 2 above)
Comment (daytonwilliams-okta) [reply packages/@okta/vuepress-site/docs/reference/api/users/index.md:741]: Slightly more terse to avoid one this one row ballooning in length.

**Review by donwoods-okta — COMMENTED** [review]
I see a second commit but it did not address any of my comments, so I'm still waiting.

File: packages/@okta/vuepress-site/docs/reference/api/users/index.md [hunk 4]
```diff
@@ -729,7 +729,28 @@ Fetches a user from your Okta organization
...
+The performance optimization will only be applied when all three parameters are passed.  Unrecognized parameters are ignored.
+
+###### Content-Type Header Examples
+
+**Header:** `Content-Type: application/json; okta-response=omitCredentials,omitCredentialsLinks`
+**Result:** Omits the credentials subobject and credentials links from the response.  Does not apply the performance optimization.
+
+
+**Header:** `Content-Type: application/json; okta-response="omitCredentials,omitCredentialsLinks, omitTransitioningToStatus"`
+**Result:** Omits the credentials, credentials links, and `transitioningToStatus` sections of the response.  Applies the performance optimization.
```
Comment (donwoods-okta) [inline packages/@okta/vuepress-site/docs/reference/api/users/index.md:753]: `transitioningToStatus` isn't much of a section; it's a single field.  The credentials links are definitely not a section; they're part of the `_links` section.  I suggest changing "sections" to "field" (singular, referring just to `transitionToStatus`) and "of the response" to "from the response".
Same changes below, of course.

**Review by donwoods-okta — COMMENTED** [review]
Thanks for the cleanup.  (Was that the intent, by the way?  That Dayton would write the content with a first cut at organizing it, and you would clean it up?  If so I apologize to @daytonwilliams-okta for griping about him closing the JIRA without addressing my PR comments.)

**Comment by jakubvul-okta** [comment]
@donwoods-okta not the intent, but that's what happened. :)

**Review by donwoods-okta — APPROVED** [review]
🚢

**Pull request merged**